quarkus-extension/engine/qa/src/test/java/io/quarkus/arquillian/QuarkusExtension.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.quarkus.arquillian;
+
+import org.camunda.bpm.engine.cdi.test.ProcessEngineDeployment;
+import org.jboss.arquillian.container.spi.client.container.DeployableContainer;
+import org.jboss.arquillian.container.test.spi.client.protocol.Protocol;
+import org.jboss.arquillian.core.spi.LoadableExtension;
+import org.jboss.arquillian.test.spi.TestEnricher;
+
+public class QuarkusExtension implements LoadableExtension {
```

Comment:
Would it make sense to create a `CamundaQuarkusExtension` class instead of duplicating the code from the `QuarkusExtension` class?

From what I can see, the Arquillian SPI loader will register all available extensions on the classpath. This way, we can avoid duplicating Quarkus code that might change in the future.

We would only need to apply the following line:
```
builder.observer(ProcessEngineDeployment.class);
```

Note: I tested this approach locally. The `engine-cdi` tests ran successfully.

Reply:
Nice solution! 👍